commands/init.go

```diff
@@ -44,26 +58,65 @@ func withClient(fn func(c client.Client, cmd *cobra.Command, args []string) erro
 	}
 }
 
-func InitClientWithUsernameAndPassword(username, password, instanceURL string) (*model.Client4, string, error) {
+func isValidChain(chain []*x509.Certificate) bool {
+	// check all certs but the root one
+	certs := chain[:len(chain)-1]
+
+	for _, cert := range certs {
+		if _, ok := insecureSignatureAlgorithms[cert.SignatureAlgorithm]; ok {
+			return false
+		}
+	}
+	return true
+}
+
+func VerifyCertificates(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+	// loop over certificate chains
+	for _, chain := range verifiedChains {
+		if isValidChain(chain) {
+			return nil
+		}
+	}
+	return fmt.Errorf("insecure algorithm found in the certificate chain. Use --insecure-sha1-intermediate flag to ignore. Aborting")
```

Comment:
I would end here the sentence with something like `Use --insecure-sha1-intermediate flag to ignore under your own responsability`, what do you think @wiersgallak?